core/app/models/spree/variant.rb

```diff
@@ -19,9 +19,10 @@ class Variant < Spree::Base
 
     # we need to have this callback before any dependent: :destroy associations
     # https://github.com/rails/rails/issues/3458
-    before_destroy :ensure_no_line_items
+    before_destroy :ensure_not_in_complete_orders
+    before_destroy :remove_line_items_from_incomplete_orders
```

Comment:
```suggestion
    after_destroy :remove_line_items_from_incomplete_orders
```

Reply:
@aleksandarpetrushev this still needs to be changed to after_destroy, not before